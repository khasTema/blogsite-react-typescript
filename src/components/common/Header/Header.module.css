.header {
    width: 100%;
    padding: 1.25rem 0;
    background-color: var(--main-color);
    margin-bottom: 2.5rem;
}

.wrapper {
    width: var(--width);
    max-width: var(--content-max-width);
    margin: 0 auto;
    display: flex;
    flex-direction: column;
    justify-content: space-between;
    align-items: center;
}

.logo {
    font-family: var(--headings-font);
    font-size: 1rem; 
}

.logo a {
    color: var(--logo-text-color);
    text-decoration: none;
}

.nav {
    display: flex;
    justify-content: space-around;
}

.link {
    display: block;
    padding: 1rem;
    border-radius: 5px;
    text-decoration: none;
    color: var(--text-color);
    font-family: var(--text-font);
    font-weight: var(--fontweight-semibold);
    transition: all .3s;
}

.link:hover {
    box-shadow: var(--nav-link-hover-shadow);
}

@media screen and (min-width: 717px) {
    .wrapper {
        flex-direction: row;
    }

    .nav {
        width: 30%;
    }

    .logo {
        font-size: 1.5rem;
    }
    
}