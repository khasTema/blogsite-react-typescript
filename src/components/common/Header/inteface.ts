export interface IheaderConfig {
    logoText: string;
    homeLink: string;
    postLink: string;
}