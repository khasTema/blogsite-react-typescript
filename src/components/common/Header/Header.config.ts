import { IheaderConfig } from "./inteface"

export const headerConfig: IheaderConfig = {
    logoText: 'Blogpost page',
    homeLink: 'Home',
    postLink: 'New Post',

}