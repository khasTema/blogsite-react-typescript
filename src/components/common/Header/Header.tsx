import React from 'react'
import { Link } from 'react-router-dom'
import styles from './Header.module.css'
import { headerConfig } from './Header.config'

export const Header: React.FC = (): JSX.Element => {
  return (
    <header className={styles.header}>
        <div className={styles.wrapper}>
            <div className={styles.logo}>
              <Link to='/'>
                <h1>{headerConfig.logoText}</h1>
              </Link>
            </div>
            <nav className={styles.nav}>
                <Link className={styles.link} to='/'>{headerConfig.homeLink}</Link>
                <Link className={styles.link} to='create/new'>{headerConfig.postLink}</Link>
            </nav>
        </div>
    </header>
  )
}
