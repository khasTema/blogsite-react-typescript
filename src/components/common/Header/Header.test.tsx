import renderer from 'react-test-renderer';
import React from 'react'
import { Header } from './Header';
import { JSX } from 'react/jsx-runtime';

jest.mock('react-router-dom');
jest.mock('./Header.module.css');
jest.mock('./Header.config');

const renderTree = (tree: JSX.Element) => renderer.create(tree);
describe('<Header>', () => {
  it('should render component Header', () => {
    expect(renderTree(
      <Header />
    ).toJSON()).toMatchSnapshot();
  });
});