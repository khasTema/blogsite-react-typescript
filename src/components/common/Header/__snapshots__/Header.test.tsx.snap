// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header> should render component Header 1`] = `
<header>
  <div>
    <div />
    <nav />
  </div>
</header>
`;
